crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -1473,25 +1473,31 @@ mod tests {
     use std::sync::Arc;
 
     use reth_chain_state::{ExecutedBlock, NewCanonicalChain};
-    use reth_primitives::B256;
-    use reth_storage_api::{BlockHashReader, BlockNumReader, HeaderProvider};
+    use reth_db::{test_utils::TempDatabase, DatabaseEnv};
+    use reth_primitives::{BlockNumberOrTag, SealedBlock, B256};
+    use reth_storage_api::{BlockHashReader, BlockNumReader, BlockReaderIdExt, HeaderProvider};
     use reth_testing_utils::generators::{self, random_block_range};
 
-    use crate::{providers::BlockchainProvider2, test_utils::create_test_provider_factory};
+    use crate::{
+        providers::BlockchainProvider2, test_utils::create_test_provider_factory, CanonChainTracker,
+    };
 
-    #[test]
-    fn test_block_hash_reader() -> eyre::Result<()> {
+    type ArcedTempDatabase = Arc<TempDatabase<DatabaseEnv>>;
+
+    const BLOCK_COUNT: usize = 10;
+
+    fn provider_with_random_blocks(
+    ) -> eyre::Result<(BlockchainProvider2<ArcedTempDatabase>, Vec<SealedBlock>)> {
```

Comment:
can inline the alias into the function return arguments, we don't use it anywhere else 
```suggestion
    const BLOCK_COUNT: usize = 10;

    fn provider_with_random_blocks(
    ) -> eyre::Result<(BlockchainProvider2<Arc<TempDatabase<DatabaseEnv>>>, Vec<SealedBlock>)> {
```